src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -303,7 +304,10 @@ public boolean authenticate(final SecurityRequestChannel request) {
                 if (authDomain.isChallenge()) {
                     final Optional<SecurityResponse> restResponse = httpAuthenticator.reRequestAuthentication(request, null);
                     if (restResponse.isPresent()) {
-                        auditLog.logFailedLogin("<NONE>", false, null, request);
+                        // saml will always hit this to re-request authentication
```

Comment:
I understand the basic problem behind this, but I am wondering whether this is the right path.

We have two challenging auth mechanisms: Basic auth and SAML. SAML will indeed always need a challenge, but also basic will cause challenges in many cases. 

What about basic auth? I have not tested it, but I THINK it will show the same behavior from Dashboards, as it will initially send an unauthenticated request to probe whether authentication is necessary.

We are here in the block guarded by the condition `if (ac == null)` in line 298. That means that no credentials have been sent. Thus, the authentication does not fail because credentials (like user name and password or SAML response) were wrong, but just because the lack of credentials.

Should this really counted then as a failed login? After all, this method won't work for brute-forcing. Maybe the call to `auditLog.logFailedLogin()` should be removed in any case?

Reply:
IMO it should be retained if the initial Basic Auth Request contains wrong credentials. If a user is making a credential-less request then I don't think its necessary. For SAML its always unnecessary.

Reply:
Marking this thread as resolved. This PR solves the linked issue. We can further enhance this for basic auth in another PR to differentiate between credential-less requests vs requests sent with wrong credentials.